node/src/components/consensus/consensus_protocol.rs

```diff
@@ -227,11 +227,11 @@ pub(crate) enum ProtocolOutcome<I, C: Context> {
     /// No progress has been made recently.
     StandstillAlert,
     /// We want to disconnect from a sender of invalid data.
-    Disconnect(I),
+    Disconnect(NodeId),
 }
 
 /// An API for a single instance of the consensus.
-pub(crate) trait ConsensusProtocol<I, C: Context>: Send {
+pub(crate) trait ConsensusProtocol<C: Context>: Send {
```

Comment:
We should be able to leave this trait parameterized with `I` and just use `NodeId` in the `era_supervisor.rs`. That way `I` is not "polluting" the code and it is abstract where we want it to be - protocol logic.

Reply:
That is definitely counter to the original assignment I was given, though I'd be happy to understand why you want it to be "abstract" there.

Reply:
Well, truth is the original assignment was pulled out of the hat by Matt and not discussed with the team :) Not saying we should not be doing that, just that the task was not discussed.

Reply:
Thank you for the context -- I'll have to think about this for a bit to form an opinion.

Reply:
Yes.

Reply:
If you're asking me @marc-casperlabs then yes - that's what I was thinking about.